cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/MonitorHandler.java

```diff
@@ -91,4 +91,36 @@ public void monitor(HttpRequest request, HttpResponder responder,
   public void getServiceSpec(HttpRequest request, HttpResponder responder) throws Exception {
     super.getServiceSpec(request, responder);
   }
+
+  /**
+   * Send request to restart all instances for a CDAP system service.
+   */
+  @Path("/system/services/{service-name}/instances/restart")
+  @PUT
+  public void restartServiceInstances(HttpRequest request, HttpResponder responder,
+                                      @PathParam("service-name") final String serviceName) {
+    super.restartServiceInstances(request, responder, serviceName);
+  }
+
+  /**
+   * Send request to restart all instances for a CDAP system service.
```

Comment:
invalid comment. should be ``restart single instance identified by <instance-id>``